GliaWidgets/Sources/EntryWidget/EntryWidget.swift

```diff
@@ -89,29 +103,48 @@ extension EntryWidget {
 
 // MARK: - Private methods
 private extension EntryWidget {
-    func handleQueuesMonitorUpdates(state: QueuesMonitor.State) {
+    func handleQueuesMonitorUpdates(state: QueuesMonitor.State, unreadSecureMessagesCount: Int?) {
         switch state {
         case .idle:
             viewState = .loading
         case .updated(let queues):
-            let availableMediaTypes = resolveAvailableMediaTypes(from: queues)
-            if availableMediaTypes.isEmpty {
+            let availableEngagementTypes = resolveAvailableEngagementTypes(from: queues)
+            if availableEngagementTypes.isEmpty {
                 viewState = .offline
             } else {
-                viewState = .mediaTypes(availableMediaTypes)
+                let mediaTypes = availableEngagementTypes.map { type in
+                    if type == .secureMessaging {
+                        return EntryWidget.MediaTypeItem(
+                            type: type,
+                            badgeCount: unreadSecureMessagesCount ?? 0
+                        )
+                    }
+                    return EntryWidget.MediaTypeItem(type: type)
+                }
+
+                viewState = .mediaTypes(mediaTypes)
             }
         case .failed:
             viewState = .error
-            print("Failed to update queues")
+            environment.log.prefixed(Self.self).warning("Failed to update queues")
```

Comment:
This one needs also to be in sync with Android logs, just in case.

Reply:
Used from spreadsheet
https://github.com/salemove/ios-sdk-widgets/pull/1112/commits/1cacfa00eac3c54059699956a407c1b4b9823ffd